### ISSUE TITLE
[feature-request] Support new Document()

### ISSUE THREAD

**Issue by ThePlenkov** [xmldom/xmldom#774]
I'd like to render Dom manually with the own logic using this library.

For that purpose I'd like to do something like:

```
import { Document } from "@xmldom/xmldom";
const document = new Document();
```

Do you think is feasible to achieve? Thanks!

**Comment by karfau** [comment]
No, this is not planned and is not part of the relevant specifications.
You need to use the methods provided by `DOMImplementation` for that purpose.
This way it is also ensured that you either create an XML or an HTML document and not something inbetween.

**Comment by karfau** [comment]
PS: The only reason to export `Document` is to access it as a type and for `instanceof` checks.

**Comment by ThePlenkov** [comment]
@karfau  thanks for a quick reply. I do not mind if I need to use DOMImplementation instead, but I'm just trying to understand.  You mentioned that is not part of relevant specs. What are specs you're referring to?

I see this spec: https://dom.spec.whatwg.org/#ref-for-dom-document-document%E2%91%A0

And here is mentioned that new Document( ) may work.  Same here : https://developer.mozilla.org/en-US/docs/Web/API/Document/Document

Thanks!

**Comment by karfau** [comment]
Thx for pointing this out.
I will have a closer look at it, but it's not very high on my priorities right now.
Which is mainly because I have not taken the time to decide what to tackle next.

We have taken the approach of preventing some constructors to be called, since they are/were(?) not listed in the specs.
I want to check which ones are listed in which versions of the spec, and potentially add them and the related types in one go.

**Comment by karfau** [comment]
Since the `XMLDocument` is basically just an alias for `Document` and doesn't add any properties or fields, it is unlikely that it will be added.
The current way of distinguishing between XML and HTML documents is the `type` property, which `DOMParser.parseFromString` derives from the mimeType parameter.

**Comment by ThePlenkov** [comment]
You're right. XMLDocument is returned only after parsing, to build XML, still Document is used: https://developer.mozilla.org/en-US/docs/Web/XML/Parsing_and_serializing_XML
